### PR TITLE
ICU-13839 Adding FormattedNumber API to PluralRules.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-13839
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/i18n/unicode/upluralrules.h [hunk 1]
```diff
@@ -148,6 +151,28 @@ uplrules_select(const UPluralRules *uplrules,
...
+ * Given a number as a UFormattedNumber, returns the keyword of the first rule
+ * that applies to the number, according to the supplied UPluralRules object.
+ *
+ * A UFormattedNumber allows you to specify an exponent or trailing zeros,
+ * which can affect the plural category. To get a FormattedNumber, see
+ * {@link UNumberFormatter}.
+ *
+ * @param uplrules The UPluralRules object specifying the rules.
+ * @param number The number for which the rule has to be determined.
+ * @param keyword The keyword of the rule that applies to number.
```
[the remark this answers is not in the document]
Comment (sffc) [reply icu4c/source/i18n/unicode/upluralrules.h]: I pushed a commit improving the documentation.

**Comment by sffc** [comment]
I moved FormattedNumber from number_fluent.cpp to a new file named number_output.cpp (for lack of a better name), and made a new lightweight dependency group named number_output.

**Comment by sffc** [comment]
@markusicu does the new dependency group look okay?

**Comment by sffc** [comment]
Quick link to the dependencies.txt: https://github.com/unicode-org/icu/pull/335/files#diff-525cbdd4b9ee84aae57d44ca0ce37389

Andy has already reviewed the rest of the code.  I don't intend for you to get caught up in the rest of the review, unless that is something you want to do.

File: icu4c/source/i18n/number_output.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+#include "unicode/numberformatter.h"
+#include "number_utypes.h"
+#include "util.h"
+#include "number_decimalquantity.h"
+
+U_NAMESPACE_BEGIN
+namespace number {
+
+
+FormattedNumber::FormattedNumber(FormattedNumber&& src) U_NOEXCEPT
```
Comment (markusicu) [inline icu4c/source/i18n/number_output.cpp]: optional: why not call this file formattednumber.cpp ?

File: icu4c/source/test/depstest/dependencies.txt [hunk 3]
```diff
@@ -937,6 +937,16 @@ group: number_representation
     ucase uniset_core
     formatted_value
 
+group: number_output
```
Comment (markusicu) [inline icu4c/source/test/depstest/dependencies.txt]: lgtm

File: icu4c/source/i18n/number_output.cpp (hunk 2 above)
Comment (sffc) [reply icu4c/source/i18n/number_output.cpp]: I name the files with the `number_` prefix.

- `number_formatted` was too close to `number_formatimpl`
- `number_formattednumber` was a bit long

so I went with `number_output` for lack of a better option.

**Comment by sffc** [comment]
^ That was the squash.  Now I need to rebase onto master in order to resolve the merge conflict.

**Comment by sffc** [comment]
^ That was the rebase.

Back to @aheninger for final review.

**Pull request merged**